TROUBLESHOOT.md

```diff
@@ -0,0 +1,230 @@
+resolve the provided SQL expressions (e.g. ">", "<", etc) while using the SQL mode to 
+read records. Users are requested to provide SQL queries containing only the 
+[supported expressions](https://github.com/GoogleCloudDataproc/flink-bigquery-connector/blob/ae7950613190b2d878bac736331ebb0032fd4d1f/flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/table/restrictions/BigQueryRestriction.java#L54).
+
+- `Can't read query results without setting a SQL query.` or 
+`Can't read query results without setting a GCP project.` - Thrown by 
+[BigQuerySource](flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/source/BigQuerySource.java) 
+when reading records from a query (via Datastream API) in case the SQL Query or GCP Project is not provided.
+Please provide the SQL Query (this is the query to be executed) and GCP Project 
+(project which runs the query) to read records from a query.
+
+ ##### Sink
+- `Error while writing to BigQuery` - Thrown by the Sink Writer in case it is unable to 
+write records to BigQuery. 
+This error is thrown when the API responsible for 
+writing records (Default Stream in at-least-once mode and Buffered Stream in 
+exactly once mode `append()` request) fails and for 
+exactly-once when response has an error or offset mismatch.
+Users are requested to consult the 
+[Java Documentation](https://cloud.google.com/java/docs/reference/google-cloud-bigquerystorage/3.5.0/com.google.cloud.bigquery.storage.v1.Exceptions) 
+for various exceptions thrown by the Storage Write APIs.
+
+- `Unable to connect to BigQuery` - Thrown by the Sink Writer
+  in case it is unable to form the Write Client while attempting to write records to BigQuery.
+Failing of StreamWriter creation causes this error. 
+Ensure proper auth permissions, correct credentials,
+internet connectivity along with suitable table schema to prevent this error.
+
+- `Commit operation failed` - Thrown by the Commiter(part of Flink's two-phase commit protocol) 
+in case it is unable to form the Write Client 
+while attempting to add rows to BQ via Storage Write API.
+Ensure proper auth permissions, correct credentials, 
+and internet connectivity to prevent this error.
+
+- `Could not obtain Descriptor from Descriptor Proto` - 
+Thrown by the [BigQuerySchemaProvider](flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/serializer/BigQuerySchemaProviderImpl.java) 
+when unable to convert the Descriptor Proto to Descriptor. 
+This error might arise when building descriptors fails because the source DescriptorProto is not a
+valid Descriptor. 
+Since the connector handles formation of DescriptorProto (for writing Avro Records to Proto), 
+users should not face this error.
+Users might face this error in case custom serializer is used.
+
+## Known Issues/Limitations
```

Comment:
let's also reference the 100 max parallelism here

Reply:
We have mentioned this in the [Readme](https://github.com/GoogleCloudDataproc/flink-bigquery-connector/blob/5462befcdd661ce1474ec6649808876df2a0f0b5/README.md?plain=1#L237), should we mention it here as well?